crates/hir-ty/src/consteval.rs

```diff
@@ -406,19 +395,24 @@ pub fn unknown_const_as_generic(ty: Ty) -> GenericArg {
 }
 
 /// Interns a constant scalar with the given type
-pub fn intern_scalar_const(value: ConstScalar, ty: Ty) -> Const {
+pub fn intern_const_scalar_with_type(value: ConstScalar, ty: Ty) -> Const {
     ConstData { ty, value: ConstValue::Concrete(chalk_ir::ConcreteConst { interned: value }) }
         .intern(Interner)
 }
 
 /// Interns a possibly-unknown target usize
-pub fn usize_const(value: Option<u64>) -> Const {
-    intern_scalar_const(
-        value.map(ConstScalar::Usize).unwrap_or(ConstScalar::Unknown),
+pub fn usize_const(value: Option<u128>) -> Const {
+    intern_const_scalar_with_type(
+        value.map(ConstScalar::UInt).unwrap_or(ConstScalar::Unknown),
         TyBuilder::usize(),
     )
 }
 
+/// Interns a constant scalar with the default type
```

Comment:
I'm not sure this function is useful; the one use below I think is dubious. Let's remove it?